da/local/local.go

```diff
@@ -94,7 +96,11 @@ func (m *DataAvailabilityLayerClient) SubmitBatch(batch *types.Batch) da.ResultS
 		return da.ResultSubmitBatch{BaseResult: da.BaseResult{Code: da.StatusError, Message: err.Error(), Error: err}}
 	}
 
-	atomic.StoreUint64(&m.daHeight, daHeight+1)
+	ok := m.daHeight.CompareAndSwap(daHeight, daHeight+1)
```

Comment:
why u need it? the entire submit process is mutex locked isn't it?

Reply:
fair enough 👍 
fixed